### PR TITLE
Added tests of encode() functions

### PR DESCRIPTION
**Pull request by mickaelandrieu**
With this sample, I dont meet any issue ... let's travis check all PhantomJS versions :)

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
@istr hmmm my tests pass ... how to fix  #777 ? :-)

**Comment by istr** [comment]
@mickaelandrieu Strange, indeed. But the test is nice; I would like to merge this. @BIGjuevos any idea why this test does not reveal the problem in #777?

**Comment by BIGjuevos** [comment]
Is it the combination of the non-english and english characters adjacent to each other?

It could also be a false positive if you copy-pasted from github into the test and the CSV from the same source rather than natively generating the characters from within code.  Character encoding is strange in my experience, and tends to behave when copy pasting.

**Comment by mickaelandrieu** [comment]
> It could also be a false positive if you copy-pasted from github into the test and the CSV from the same source rather than natively generating the characters from within code. 

I did :p any idea on how I can generate this kind of characters within code ?

**Comment by BIGjuevos** [comment]
> I did :p any idea on how I can generate this kind of characters within code ?

On mac, go into keyboard preferences and add another keyboard layout that is a troublesome layout, and mash some keys to get it directly inputted.

File: tests/suites/casper/encode.js [hunk 1]
```diff
@@ -12,6 +12,29 @@ casper.test.begin('base64encode() and download() tests', 2, function(test) {
...
+
+        var stream = fs.open('__base64encode.csv', 'r'),
+        expectedValues = [
+            'إختبار,',
+            'آزمایشی,',
+            '测试',
+            'испытание',
+            'परीकi'
+        ],
+        i = 0,
```
Comment (mickaelandrieu) [inline tests/suites/casper/encode.js]: as a note for myself: should be `var i = 0;`

**Comment by jefleponot** [comment]
it is quite strange to me to add this kind of test but code seems ok.
agree to merge

**Comment by mickaelandrieu** [comment]
@jefleponot we try to cover some edge errors but the test don't cover all the cases, according to @BIGjuevos comments :/

**Comment by mickaelandrieu** [comment]
Note I have never reproduced this issue, so it's hard for me to write an efficient test :)

**Comment by jefleponot** [comment]
@mickaelandrieu , are you ok to merge this PR ?

**Comment by mickaelandrieu** [comment]
I can't validate my own pull requests :cat:

**Comment by jefleponot** [comment]
lol, I'm ok for this PR. I just want to know if you still want that I add your PR.

**Comment by mickaelandrieu** [comment]
Yes ! :+1:

**Pull request merged**